Notebook to run estimate

sjspielman opened the pull request:
Closes #154 

This PR adds a notebook to run ESTIMATE on all data quantities and visualize and export results. I made several comparison plots, and tried to dig a little further into the patterns I saw for one project in particular (with "tried" being the operative word here). Let me know if you can think of anything else that belongs in the notebook at this time? Any additional comparisons that you'd like to see or shifts in plotting strategies? Here's the rendered notebook:
[run-estimate.nb.html.zip](https://github.com/user-attachments/files/18869670/run-estimate.nb.html.zip)

Note that `renv.lock` was updated to get `tidyestimate`, for which a few dependencies also were updated too. FYI also - as part of pre-planning next sprint, I'll present the results from these analyses more generally for broader group discussion about next steps.

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
```diff
@@ -0,0 +1,289 @@
+
+We will now run ESTIMATE on all three types of counts for samples in each project:
+
+* bulk counts
+* bulk TPM
+* pseudobulk counts
```
Can you include brief descriptions of what these are? I know we've been using them in other analyses, but I think it would be helpful context when just looking at this notebook. Mainly I just want to be explicit that bulk counts and TPM come from the bulk RNA-seq data and pseudobulk counts come from the single-cell RNA seq data.

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
```diff
@@ -0,0 +1,289 @@
+p2 <- ggplot(project1_df) + 
+  aes(x = technology, y = pseudobulk) + 
+  geom_jitter(width = 0.1) + 
+  ggtitle("pseudobulk ~ technology")
```
I don't know that with pseudobulk looking at single-end and paired-end makes a whole lot of sense? That really shouldn't affect the pseudobulk samples at all since that information is only relevant for bulk.

sjspielman replied:
Ah yes! definitely correct.

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
```diff
@@ -0,0 +1,289 @@
+* There does not appear to be a significant relationship between modalities for SCPCP000002 and SCPCP000006, for either score 
+* For SCPCP000001, immune scores tend to be higher in pseudobulk but stromal scores are about the same between modalities
+* For SCPCP000006 and SCPCP0000017, immune scores tend to be lower in pseudobulk but stromal scores are about the same between modalities
+* Note that projects SCPCP000001 and SCPCP000002 are single-cell, but the other three projects are single-nucleus
```
It might be worth a plot where you just look at immune and stromal scores in pseudobulk and group by `seq_unit`

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
````diff
@@ -0,0 +1,289 @@
+  theme(aspect.ratio = 1)
+```
+
+* There does not appear to be a significant relationship between modalities for SCPCP000002 and SCPCP000006, for either score 
````
This is true, but in SCPCP000002 we see higher scores across the board in pseudobulk which I think is interesting.

allyhawkins replied:
Hmm how are you defining a "significant" relationship here? Visually the only one that doesn't seem to have one is 2, but 6 appears to have some?

sjspielman replied:
Aha, this is a typo! Should have said 9, not 6.

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
```diff
@@ -0,0 +1,289 @@
+## Visualization
+
+
+Let's begin by looking at the overall values:
```
I might switch the order of this notebook a little bit. 
- first look at the relationship between bulk TPM and counts 
- then pick one and show this sina plot for just the counts and pseudobulk followed by the comparison of counts to pseudobulk.

allyhawkins reviewed (commented):
This generally looks good and most of my comments are around organization. One thing that I do think is interesting that is missing is a comparison between the immune score from pseudobulk and % of cells that are classified as immune in the consensus cell types. I would hope that those are somewhat concordant, but I'm also thinking that should be a separate notebook once we bring the consensus cell typing into this repo. 

I'm going to go out on a limb here and say that generally I like this analysis and I think there are observable differences in all projects that we could display.

sjspielman commented:
> One thing that I do think is interesting that is missing is a comparison between the immune score from pseudobulk and % of cells that are classified as immune in the consensus cell types. I would hope that those are somewhat concordant, but I'm also thinking that should be a separate notebook once we bring the consensus cell typing into this repo.

I _definitely_ do want to be looking at this!! Just didn't think we quite had all the data in place for it in this repo yet, but maybe next sprint?

jaclyn-taroni commented:
> Just didn't think we quite had all the data in place for it in this repo yet, but maybe next sprint?

Is there anything preventing you from grabbing the labels from the workflow results bucket to start?

sjspielman commented:
> Is there anything preventing you from grabbing the labels from the workflow results bucket to start?

sjspielman commented:
@allyhawkins for this PR, I'll plan to update organization as you suggest (fyi probably tomorrow since i'm writing my self-eval this afternoon). After that, I'll add a separate notebook comparing to consensus which I'll grab from S3 to for now.

sjspielman commented:
As requested, I've rearranged the notebook, updated some text, added the additional seq_unit plot, and removed the silly technology plot. 
[run-ESTIMATE.nb.html.zip](https://github.com/user-attachments/files/18888919/run-ESTIMATE.nb.html.zip)

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
```diff
@@ -0,0 +1,317 @@
+
+#### Compare across sequencing unit
+
+On the single-cell level, some projects used single-cell libraries and others used single-nucleus libraries. 
```
This doesn't surprise me one bit and it's a nice validation to see that single-cell you get more immune cells than single-nuclei!

allyhawkins commented on analysis/estimate/notebooks/run-estimate.Rmd:
````diff
@@ -0,0 +1,317 @@
+#### Compare modalities
+
+```{r fig.width=10}
+compare_modalities_df <- estimate_df |>
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  ) 
````
I think you do this twice and can remove this?

sjspielman replied:
well it sure sounds like something I'd do 😂

allyhawkins reviewed (commented):
This looks good. I have one lingering comment about interpretation that should probably be addressed before this goes in.

sjspielman commented:
I removed the duplicated data frame, and fixed that typo as it is project 9, not 6, that has no relationship.

allyhawkins reviewed (approved):
LGTM
